Modules/EMCAL/src/RawTask.cxx

```diff
@@ -0,0 +1,278 @@
+    mRawAmplitudeEMCAL[i]->GetYaxis()->SetTitle("Counts");
+    getObjectsManager()->startPublishing(mRawAmplitudeEMCAL[i]);
+
+    mRMSperSM[i] = new TProfile2D(Form("RMSADCperSM%d", i), Form("RMSperSM%d", i), 48, 0, 48, 24, 0, 24);
+    mRMSperSM[i]->GetXaxis()->SetTitle("col");
+    mRMSperSM[i]->GetYaxis()->SetTitle("row");
+    getObjectsManager()->startPublishing(mRMSperSM[i]);
+
+    mMEANperSM[i] = new TProfile2D(Form("MeanADCperSM%d", i), Form("MeanADCperSM%d", i), 48, 0, 48, 24, 0, 24);
+    mMEANperSM[i]->GetXaxis()->SetTitle("col");
+    mMEANperSM[i]->GetYaxis()->SetTitle("row");
+    getObjectsManager()->startPublishing(mMEANperSM[i]);
+
+    mMAXperSM[i] = new TProfile2D(Form("MaxADCperSM%d", i), Form("MaxADCperSM%d", i), 48, 0, 47, 24, 0, 23);
+    mMAXperSM[i]->GetXaxis()->SetTitle("col");
+    mMAXperSM[i]->GetYaxis()->SetTitle("row");
+    getObjectsManager()->startPublishing(mMAXperSM[i]);
+  }
+}
+
+void RawTask::startOfActivity(Activity& /*activity*/)
+{
+  QcInfoLogger::GetInstance() << "startOfActivity" << AliceO2::InfoLogger::InfoLogger::endm;
+  reset();
+}
+
+void RawTask::startOfCycle()
+{
+  QcInfoLogger::GetInstance() << "startOfCycle" << AliceO2::InfoLogger::InfoLogger::endm;
+}
+
+void RawTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  // In this function you can access data inputs specified in the JSON config file, for example:
+  //   "query": "random:ITS/RAWDATA/0"
+  // which is correspondingly <binding>:<dataOrigin>/<dataDescription>/<subSpecification
+  // One can also access conditions from CCDB, via separate API (see point 3)
+
+  // Use Framework/DataRefUtils.h or Framework/InputRecord.h to access and unpack inputs (both are documented)
+  // One can find additional examples at:
+  // https://github.com/AliceO2Group/AliceO2/blob/dev/Framework/Core/README.md#using-inputs---the-inputrecord-api
+
+  using CHTYP = o2::emcal::ChannelType_t;
+  // Some examples:
+
+  // 1. In a loop
+  for (auto&& input : ctx.inputs()) {
+    // get message header
+    if (input.header != nullptr && input.payload != nullptr) {
+      const auto* header = header::get<header::DataHeader*>(input.header);
+      // get payload of a specific input, which is a char array.
+      // const char* payload = input.payload;
+
+      //fill the histogram with payload sizes
+      mHistogram->Fill(header->payloadSize);
+
+      // try decoding payload
+      o2::emcal::RawReaderMemory<o2::header::RAWDataHeaderV4> rawreader(gsl::span(input.payload, header->payloadSize));
+      uint64_t currentTrigger(0);
+      bool first = true; //for the first event
+      short int maxADCSM[20];
+      while (rawreader.hasNext()) {
+        rawreader.next();
+        auto payLoad = rawreader.getPayload();
+        auto payLoadSizeTest = payLoad.getPayloadWords().size() * sizeof(uint32_t); //payloadsize in byte
+
+        auto headerR = rawreader.getRawHeader();
+        mPayloadSizePerDDL->Fill(headerR.feeId, payLoadSizeTest / 1024.);
```

Comment:
The PR https://github.com/AliceO2Group/AliceO2/pull/2710 has just been merged. One can now simply get the payload size via  `rawreader.getPayloadSize()`.

Reply:
Thanks Markus for your comments! I have updated the code with your suggestions!